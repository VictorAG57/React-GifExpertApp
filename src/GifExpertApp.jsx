
import React, {Fragment} from 'react';
import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categorys, setCategory] = useState(["Rick & Morty"]);

    // const addCategory = () => {
    //     setCategoria(
    //         [...categorys, "Family guy"]
    //     )
    // }

    return (
        <Fragment>
            <h2>GifExpertApp</h2>

            {/* 
                Mediante los props podemos enviar también funciones
                en este caso enviamos nuestra función para poder cambiar el estado
                "setCategory".
            */}
            <AddCategory setCategory={ setCategory } ></AddCategory>
            
            <hr></hr>

            <ol>
                {
                    categorys.map( category =>( 
                        //return <li key={ category }> { category } </li>
                        <GifGrid key={ category } category={ category } ></GifGrid>
                    ))
                }
            </ol>
        </Fragment>
    )
}

export default GifExpertApp;
