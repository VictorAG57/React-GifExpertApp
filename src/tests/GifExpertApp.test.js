import GifExpertApp from '../GifExpertApp';
import { shallow } from 'enzyme';

describe("Debe de mostrar el componente correctamente", ()=> {

    const title = "Un titulo"
    const url = "https://titulo.png"
    
    test("Mostrar el componente", ()=> {

        const wrapper = shallow(<GifExpertApp></GifExpertApp>)
        expect(wrapper).toMatchSnapshot()
    });

    test('Debe mostrar una lista de categorias', () => {

        const categorys = ['Rick and Morty'];
        const wrapper = shallow(<GifExpertApp defaultCategorys={ categorys }></GifExpertApp>)

        // Hacemos un snapShot y lo evaluamos
        expect( wrapper ).toMatchSnapshot();
        
        // Nuestro GifExpertApp debe contener dentro la llamada de un componete llamado 'GifGrid'
        // y aquí evaluamos cuantos elementos tiene dentro (1 elemento:category), 
        // que deben ser los mismos que nuestra categoría
        // que es la que le enviamos a dicho conponente
        expect( wrapper.find('GifGrid').length ).toBe( categorys.length )
        
    });
}); 