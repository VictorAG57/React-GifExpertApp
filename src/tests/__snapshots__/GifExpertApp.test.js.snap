// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe de mostrar el componente correctamente Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category={
        Array [
          "Rick and Morty",
        ]
      }
      key="Rick and Morty"
    />
  </ol>
</Fragment>
`;

exports[`Debe de mostrar el componente correctamente Mostrar el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Rick and Morty"
      key="Rick and Morty"
    />
  </ol>
</Fragment>
`;
